src/garage/torch/policies/gaussian_mlp_policy.py

```diff
@@ -0,0 +1,139 @@
+                'softplus' : softplus value
+        max_std (float): If not None, the std is at most the value of max_std,
+            to avoid numerical issues.
+            > std_parameterization
+                'exp' : plain value (not log or exponentiated)
+                'softplus' : softplus value
+        std_hidden_nonlinearity: Nonlinearity for each hidden layer in
+            the std network.
+        std_output_nonlinearity: Nonlinearity for output layer in
+            the std network.
+        std_parametrization (str): How the std should be parametrized. There
+            are a few options:
+        - exp: the logarithm of the std will be stored, and applied a
+            exponential transformation
+        - softplus: the std will be computed as log(1+exp(x))
+        layer_normalization (bool): Bool for using layer normalization or not.
+    :return:
+
+    """
+    def __init__(self,
+                 env_spec,
+                 hidden_sizes=(32, 32),
+                 hidden_nonlinearity=torch.tanh,
+                 hidden_w_init=nn.init.xavier_uniform_,
+                 hidden_b_init=nn.init.zeros_,
+                 output_nonlinearity=None,
+                 output_w_init=nn.init.xavier_uniform_,
+                 output_b_init=nn.init.zeros_,
+                 learn_std=True,
+                 adaptive_std=False,
+                 std_share_network=False,
+                 init_std=1.0,
+                 min_std=1e-6,
+                 max_std=None,
+                 std_hidden_sizes=(32, 32),
+                 std_hidden_nonlinearity=torch.tanh,
+                 std_output_nonlinearity=None,
+                 std_parameterization='exp',
+                 layer_normalization=False):
+        nn.Module.__init__(self)
+        Policy.__init__(self, env_spec)
+
+        obs_dim = env_spec.observation_space.flat_dim
+        action_dim = env_spec.action_space.flat_dim
+
+        self._model = GaussianMLPModule(
+            input_dim=obs_dim,
+            output_dim=action_dim,
+            hidden_sizes=hidden_sizes,
+            hidden_nonlinearity=hidden_nonlinearity,
+            hidden_w_init=hidden_w_init,
+            hidden_b_init=hidden_b_init,
+            output_nonlinearity=output_nonlinearity,
+            output_w_init=output_w_init,
+            output_b_init=output_b_init,
+            learn_std=learn_std,
+            adaptive_std=adaptive_std,
+            std_share_network=std_share_network,
+            init_std=init_std,
+            min_std=min_std,
+            max_std=max_std,
+            std_hidden_sizes=std_hidden_sizes,
+            std_hidden_nonlinearity=std_hidden_nonlinearity,
+            std_output_nonlinearity=std_output_nonlinearity,
+            std_parameterization=std_parameterization,
+            layer_normalization=layer_normalization
+        )
+
+    def forward(self, inputs):
+        return self._model(inputs)
+
+    def get_action(self, observation):
```

Comment:
this has a very predictable implementation for pretty much all vectorized policies -- can it just be in a base class?

```python
def get_action(self, observation):
    return np.squeeze(self.get_actions(np.expand_dims(observation, axis=0)))
```

Reply:
Currently, I do opposite way

`
    def get_actions(self, observations):
        """Get actions given observations."""
        return np.stack([self.get_action(obs) for obs in observations])
`

do you think we need to change this code?

Reply:
yes -- you want to use a vectorized feed to get_actions, not a for loop

@krzentner will torch do this automatically or does he need to feed a batch?

Reply:
fixed